### DISCUSSION TITLE
After using Actor::kill, is it still necessary to call JoinHandle.await?

### DISCUSSION THREAD

**Discussion by phil-skillwon** [slawlor/ractor#174] [Q&A] [answered]
When I exit a actor, I first call ActorCell::stop, then call JoinHandle.await. Suppose I didn't save JoinHandle, can I directly use ActorCell::kill? 
In other words:   After using Actor::kill, is it still necessary to call JoinHandle.await?
Could you please explain in detail? Thank you

**Comment by slawlor** [comment, marked as answer]
You should await the joinhandle ideally, it'll interrupt the messaging task under the hood, however there is some non-instant cleanup which needs to execute for the actor to fully die. If you don't await the handle, and immediately exit, you may have a form of memory-leak because tokio will have some pending tasks living around.

You generally don't need to wait on the handles if you're not immediately shutting down however, so in "general" you just do it on the root-process inside the `main` of your process to not exit the process too quickly.
